ygnmi/gnmi.go

```diff
@@ -330,25 +333,49 @@ const (
 )
 
 // populateSetRequest fills a SetResponse for a val and operation type.
-func populateSetRequest(req *gpb.SetRequest, path *gpb.Path, val interface{}, op setOperation, preferShadowPath bool) error {
+func populateSetRequest(req *gpb.SetRequest, path *gpb.Path, val interface{}, op setOperation, preferShadowPath bool, opts ...Option) error {
 	if req == nil {
 		return fmt.Errorf("cannot populate a nil SetRequest")
 	}
+	opt := resolveOpts(opts)
 
 	switch op {
 	case deletePath:
 		req.Delete = append(req.Delete, path)
 	case replacePath, updatePath:
-		// Since the GoStructs are generated using preferOperationalState, we
-		// need to turn on preferShadowPath to prefer marshalling config paths.
-		js, err := ygot.Marshal7951(val, ygot.JSONIndent("  "), &ygot.RFC7951JSONConfig{AppendModuleName: true, PreferShadowPath: preferShadowPath})
-		if err != nil {
-			return fmt.Errorf("could not encode value into JSON format: %w", err)
-		}
 		update := &gpb.Update{
 			Path: path,
-			Val:  &gpb.TypedValue{Value: &gpb.TypedValue_JsonIetfVal{JsonIetfVal: js}},
 		}
```

Comment:
It seems to me it's cleaner to put this at the end. There seems to be multiple overwritings of update.Val, which makes it a bit more difficult to read.